core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -47,6 +53,27 @@ class NettyBlockTransferService(conf: SparkConf, securityManager: SecurityManage
   private[this] var server: TransportServer = _
   private[this] var clientFactory: TransportClientFactory = _
   private[this] var appId: String = _
+  
+  private[this] var clock: Clock = new SystemClock()
```

Comment:
I think you could add a new method here like in [ExecutorAllocationManager](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala#L209) to let user to set ManualClock, this will be easy for test.